orc/src/main/java/org/apache/iceberg/orc/ColumnMap.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.orc;
+
+import com.google.common.base.Splitter;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.orc.TypeDescription;
+
+/**
+ * The mapping from ORC's TypeDescription to the Iceberg column ids.
+ * <p>
+ * Keep the API limited to Map rather than a concrete type so that we can
+ * change it later.
+ */
+final class ColumnMap {
```

Comment:
I don't think this needs to be `final`, but it should be fine. Also, is there a better name than `ColumnMap` now that includes only static util methods? What about `ORCSchemaUtil` to match the conventions used in Avro and Parquet?